### PR TITLE
[bitnami/nginx]: Fix template scope for .Values

### LINKED ISSUE #22474: bitnami/nginx helm templating error
### Name and Version

bitnami/nginx 15.8.1

### What architecture are you using?

None

### What steps will reproduce the bug?

Install bitnami/nginx without setting `containerPorts.https`.

### Are you using any custom parameters or values?

_No response_

### What is the expected behavior?

Chart installed.

### What do you see instead?

```
Error: template: nginx/templates/ingress.yaml:54:164: executing "nginx/templates/ingress.yaml" at <.Values.containerPorts.https>: nil pointer evaluating interface {}.containerPorts
```

### Additional information

In #20934 a templating error was introduced; `.Values` is not available within a `range` loop.

### PR DESCRIPTION
**Pull request by robinelfrink**
### Description of the change

Fix Helm templating error.

### Benefits

Helm Chart bitnami/nginx can be installed with `ingress.extraHosts` set.

### Applicable issues

- fixes #22474

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/nginx/templates/ingress.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ spec:
             {{- if eq "true" (include "common.ingress.supportsPathType" .) }}
             pathType: {{ .Values.ingress.pathType }}
             {{- end }}
-            backend: {{- include "common.ingress.backend" (dict "serviceName" (include "common.names.fullname" .) "servicePort" (ternary "http" "https" (not .Values.containerPorts.https)) "context" $)  | nindent 14 }}
+            backend: {{- include "common.ingress.backend" (dict "serviceName" (include "common.names.fullname" .) "servicePort" (ternary "http" "https" (not $.Values.containerPorts.https)) "context" $)  | nindent 14 }}
```
Comment (FraPazGal) [inline bitnami/nginx/templates/ingress.yaml:33]: Hi @robinelfrink! Unless I'm missing something, this line is not inside a `range` and doesn't need the context change right?

Comment (fmulero) [reply bitnami/nginx/templates/ingress.yaml:33]: This is not needed

**Review by fmulero — CHANGES_REQUESTED** [review]
Please @robinelfrink  check health-ingress also https://github.com/bitnami/charts/blob/1c685b3216cb11d3224d5bf8b6ca9cba9b2d6175/bitnami/nginx/templates/health-ingress.yaml#L43

File: bitnami/nginx/templates/ingress.yaml (hunk 1 above)
Comment (robinelfrink) [reply bitnami/nginx/templates/ingress.yaml]: No, it isn't. Thanks.

**Comment by robinelfrink** [comment]
> Please @robinelfrink check health-ingress also

`health-ingress` is fine as it is. Tested with values:

```
image:
  debug: true
service:
  type: ClusterIP
ingress:
  enabled: true
  annotations:
    nginx.ingress.kubernetes.io/whitelist-source-range: "0.0.0.0/0"
  ingressClassName: nginx
  hostname: "my-nginx.xxx"
  extraHosts:
  - name: "another-host.xxx"
healthIngress:
  enabled: true
```

**Comment by FraPazGal** [comment]
That is weird @robinelfrink, as my colleague said, the `health-ingress` template seems to be in the same situation with a wrong context being used inside the range. See it uses `.Values.containerPorts`: 
https://github.com/bitnami/charts/blob/fd20c73fcd7231550b12dc837fa35c4911de5fc0/bitnami/nginx/templates/health-ingress.yaml#L35-L44

You are not using the `.Values.healthIngress.extraHosts` in your example above so that may explain why it didn't fail on your side:

```
$ cat test.yaml
healthIngress:
  enabled: true
  extraHosts:
  - name: "another-host.xxx"
$ helm template nginx bitnami/nginx -f test.yaml
Error: template: nginx/templates/health-ingress.yaml:43:164: executing "nginx/templates/health-ingress.yaml" at <.Values.containerPorts.https>: nil pointer evaluating interface {}.containerPorts
```

**Comment by robinelfrink** [comment]
> You are not using the `.Values.healthIngress.extraHosts` in your example above so that may explain why it didn't fail on your side:

I missed that completely.

**Review by fmulero — APPROVED** [review]
LGTM. Thanks a lot @robinelfrink

**Pull request merged**